openupgrade_scripts/scripts/base/15.0.1.3/pre-migration.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2020 Odoo Community Association (OCA)
+# Copyright 2020 Opener B.V. <stefan@opener.am>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import logging
+
+from openupgradelib import openupgrade
+
+from odoo import tools
+
+_logger = logging.getLogger(__name__)
+
+try:
+    from odoo.addons.openupgrade_scripts.apriori import merged_modules, renamed_modules
+except ImportError:
+    renamed_modules = {}
+    merged_modules = {}
+    _logger.warning(
+        "You are using openupgrade_framework without having"
+        " openupgrade_scripts module available."
+        " The upgrade process will not work properly."
+    )
+
+
+@openupgrade.migrate(use_env=False)
+def migrate(cr, version):
+    """
+    Don't request an env for the base pre migration as flushing the env in
+    odoo/modules/registry.py will break on the 'base' module not yet having
+    been instantiated.
+    """
+    if "openupgrade_framework" not in tools.config["server_wide_modules"]:
+        logging.error(
+            "openupgrade_framework is not preloaded. You are highly "
+            "recommended to run the Odoo with --load=openupgrade_framework "
+            "when migrating your database."
+        )
```

Comment:
It seems like you will need something like this:

```suggestion
        )
        openupgrade.rename_xmlids(cr, [("mail.icp_mail_catchall_alias", "base.icp_mail_catchall_alias")])
```